Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/CosmosServerlessSparkBatchJob.java

```diff
@@ -152,7 +146,13 @@ public String getJobUuid() {
     @Override
     public Observable<SparkBatchJobResponsePayload> getStatus() {
         return getAccount().getSparkBatchJobRequest(getJobUuid())
-                .map(sparkBatchJob -> sparkBatchJob.properties().responsePayload());
+                .flatMap(sparkBatchJob -> {
+                    if (sparkBatchJob.properties() == null) {
+                        return Observable.empty();
```

Comment:
Should we return `Observable.error()`? Is it null properties acceptable?

Reply:
`Observable.error()` will trigger killing the spark job, I'm afraid this is not what we want.

---
In reply to: [244076924](https://github.com/Microsoft/azure-tools-for-java/pull/2530#discussion_r244076924) [](ancestors = 244076924)